### ISSUE TITLE
aerich 0.4.1 init-db and the missing ##### downgrade ##### section on the file

### ISSUE THREAD

**Issue by IgooorGP** [tortoise/aerich#96]
Hi there, @long2ice !

First, thanks for the awesome project, I have been using on a playground project of [mine](https://github.com/IgooorGP/telegraph) and having fun with it 🥳 !

Two days ago I tried to update `aerich` from version `0.3.3` to [0.4.0](https://github.com/tortoise/aerich/releases/tag/v0.4.0) and had some issues that were fixed by [0.4.1](https://github.com/tortoise/aerich/releases/tag/v0.4.1) bugfix (same error mentioned [here](https://github.com/tortoise/aerich/issues/93)).

While `0.4.1` fixed that bug, when using the new `SQL migrations` introduced by `0.4.0` with `aerich init-db`, I'm having problems at this part of [aerich's code](https://github.com/tortoise/aerich/blob/dev/aerich/utils.py#L68) where the migration file is opened and the section `##### downgrade #####/##### upgrade #####` are looked for:

```python
Traceback (most recent call last):
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/bin/aerich", line 8, in <module>
    sys.exit(main())
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/aerich/cli.py", line 289, in main
    cli()
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/aerich/cli.py", line 35, in wrapper
    loop.run_until_complete(f(*args, **kwargs))
  File "/usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/aerich/cli.py", line 113, in upgrade
    content = get_version_content_from_file(file_path)
  File "/Users/igp/.local/share/virtualenvs/telegraph-uVCFQkwG/lib/python3.9/site-packages/aerich/utils.py", line 68, in get_version_content_from_file
    second = content.index(_DOWNGRADE)
ValueError: substring not found
```

This is the initial SQL migration generated and as it's the first migration, the section `##### downgrade #####` is not created (just like in the old jsons):

```sql
##### upgrade #####
CREATE TABLE IF NOT EXISTS "chatroom" (
    "id" UUID NOT NULL  PRIMARY KEY,
    "created_at" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,
    "updated_at" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,
    "name" VARCHAR(255) NOT NULL,
    "description" VARCHAR(255) NOT NULL,
    "max_concurrent_users" INT NOT NULL,
    "is_active" BOOL NOT NULL  DEFAULT True
);
CREATE INDEX IF NOT EXISTS "idx_chatroom_created_d59897" ON "chatroom" ("created_at");
CREATE INDEX IF NOT EXISTS "idx_chatroom_updated_1d41e3" ON "chatroom" ("updated_at");

...

CREATE TABLE IF NOT EXISTS "aerich" (
    "id" SERIAL NOT NULL PRIMARY KEY,
    "version" VARCHAR(255) NOT NULL,
    "app" VARCHAR(20) NOT NULL,
    "content" TEXT NOT NULL
);;
```

Subsequent model changes do actually add both sections, for example:

```sql
##### upgrade #####
ALTER TABLE "chatroom" ADD "new_column" INT NOT NULL;
##### downgrade #####
ALTER TABLE "chatroom" DROP COLUMN "new_column";
```

I fixed aerich's version back to `0.3.3`, there the old `json` migration file works well, even with just the `"upgrade"` key:

```json
{
  "upgrade": [
    "CREATE TABLE IF NOT EXISTS \"chatroom\" (\n    \"id\" UUID NOT NULL  PRIMARY KEY,\n    \"created_at\" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,\n    \"updated_at\" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,\n    \"name\" VARCHAR(255) NOT NULL,\n    \"description\" VARCHAR(255) NOT NULL,\n    \"max_concurrent_users\" INT NOT NULL,\n    \"is_active\" BOOL NOT NULL  DEFAULT True\n);\nCREATE INDEX IF NOT EXISTS \"idx_chatroom_created_d59897\" ON \"chatroom\" (\"created_at\");\nCREATE INDEX IF NOT EXISTS \"idx_chatroom_updated_1d41e3\" ON \"chatroom\" (\"updated_at\");\nCREATE INDEX IF NOT EXISTS \"idx_chatroom_name_f56787\" ON \"chatroom\" (\"name\");\nCREATE INDEX IF NOT EXISTS \"idx_chatroom_name_57c5fa\" ON \"chatroom\" (\"name\", \"is_active\");\nCOMMENT ON TABLE \"chatroom\" IS 'Models a chatroom.';\nCREATE TABLE IF NOT EXISTS \"user\" (\n    \"id\" UUID NOT NULL  PRIMARY KEY,\n    \"created_at\" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,\n    \"updated_at\" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,\n    \"nick_name\" VARCHAR(255) NOT NULL UNIQUE\n);\nCREATE INDEX IF NOT EXISTS \"idx_user_created_b19d59\" ON \"user\" (\"created_at\");\nCREATE INDEX IF NOT EXISTS \"idx_user_updated_dfdb43\" ON \"user\" (\"updated_at\");\nCOMMENT ON TABLE \"user\" IS 'Models a chat user.';\nCREATE TABLE IF NOT EXISTS \"message\" (\n    \"id\" UUID NOT NULL  PRIMARY KEY,\n    \"created_at\" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,\n    \"updated_at\" TIMESTAMPTZ   DEFAULT CURRENT_TIMESTAMP,\n    \"message\" VARCHAR(255) NOT NULL,\n    \"user_id\" UUID NOT NULL REFERENCES \"user\" (\"id\") ON DELETE CASCADE,\n    \"chat_room_id\" UUID NOT NULL REFERENCES \"chatroom\" (\"id\") ON DELETE CASCADE\n);\nCREATE INDEX IF NOT EXISTS \"idx_message_created_e537a8\" ON \"message\" (\"created_at\");\nCREATE INDEX IF NOT EXISTS \"idx_message_updated_db912a\" ON \"message\" (\"updated_at\");\nCOMMENT ON TABLE \"message\" IS 'Models a chat message coming from a user in a chatroom.';\nCREATE TABLE IF NOT EXISTS \"aerich\" (\n    \"id\" SERIAL NOT NULL PRIMARY KEY,\n    \"version\" VARCHAR(255) NOT NULL,\n    \"app\" VARCHAR(20) NOT NULL,\n    \"content\" TEXT NOT NULL\n);"
  ]
}
```

Have I done something wrong? Do you need any extra details to help on the troubleshoot? If so, I can provide them here. I have fixed aerich back to `0.3.3` and the project is running fine.

Thanks in advance!

**Comment by long2ice** [comment]
Yes as you see, the initial has no `downgrade` section becase there is no downgrade action, that's intentional. So what command caused the error above?

**Comment by IgooorGP** [comment]
Hi there, @long2ice ! Thanks for the reply. I spent some more time trying to understand a little bit more of the issue and the sequence of steps to reproduce it:

To begin working with my project, I ran the following commands:

```bash
docker-compose up -d db  # bring a new fresh postgresql instance for my app
aerich init-db  # initial schema and first migration execution (creates aerich migration tables on postgres)
```

After that, on every local app booting, I run the following commands:

```bash
$ aerich upgrade  # makes sure all existing migrations are applied (this step might not be really necessary)
> No migrate items

$ aerich migrate # generate new migrations  # generate new migrations if there are model changes
> No changes detected

$ aerich upgrade # applies possible new migrations
> No migrate items
```

This creates my first sql migration file (without the downgrade section as we discussed), let's say: `migration1_init.sql`. That's okay, it all runs fine and my app is good to go! If I change my models, the `aerich upgrade + aerich migrate` on the app boot happens and all the migrations go well.

Once I'm done working with the project, I usually run: `docker-compose down`, which kills my postgresql instance naturally.

Now comes the problem: to start working again on my project, I'll run again:

```bash
docker-compose up -d db  # brings up a new fresh postgres instance (without aerich migration tables or any aerich-related state)
```

And, as the first migration file already exists (the mentioned `migration1_init.sql`), my local app booting runs again:

```
aerich upgrade  # <---------- this raises the problem: ValueError: substring not found (_DOWNGRADE)
aerich migrate  # (this also raises problems if I try to run it manually)
aerich upgrade
```

It seems that as there are **no aerich tables** on the new postgres instance, this errors happens. However, I can't run `aerich init-db` again, as, well, I had already run the first migration previously (`migration1_init.sql` file already exists) and even if I did, as my `models` folder already exists, it says:

```bash
$ aerich init-db
Inited models already, or delete ./src/migrations/models and try again.
```

So, to sum up: the problem happens when I attempt to run `aerich upgrade` when **there is already an initial migration** file only (`migration1_init.sql`), it seems that the statements there will only work with `aerich init-db`, which makes sense in a way, but, shouldn't we be supposed to be able to run that initial migration file also with a simple `aerich upgrade` command (in my case, in a fresh db instance)?

Moreover, I tested the behavior with `aerich 0.3.3` and there's no such issue: if there's an `migration1_init.json` and on a new postgres instance I try to run it with `aerich upgrade`, it all goes well!

Can we make the same behavior happen with `0.4.x` versions with the sql migrations?

**Comment by long2ice** [comment]
I know, that's a bug and fixed in latest dev branch, just try that.

**Comment by IgooorGP** [comment]
Great, gonna try it after I get off from work! Thanks for all the support @long2ice .

**Comment by anikolaienko** [comment]
Hi @long2ice , any projections when this fix will be released? Thanks

**Comment by long2ice** [comment]
Released [v0.4.2](https://github.com/tortoise/aerich/releases/tag/v0.4.2)

**Comment by IgooorGP** [comment]
Thanks for the release, @long2ice ! I have just upgraded my project here, and now it works 🎉! Thanks a lot!! Feel free to close this!

**Comment by anikolaienko** [comment]
Yeay, thanks @long2ice , gonna use it now in my project as well!
